api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -51,26 +53,42 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	public Cohort getCohort(Integer id) throws DAOException {
 		return (Cohort) sessionFactory.getCurrentSession().get(Cohort.class, id);
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.CohortDAO#getCohortsContainingPatientId(java.lang.Integer)
+	 */
+	@SuppressWarnings("unchecked")
+	public List<Cohort> getCohortsContainingPatientId(Integer patientId, Boolean voided) throws DAOException {
+		Disjunction orDisjunction = Restrictions.disjunction();
+		orDisjunction.add(Restrictions.isNull("m.endDate"));
+		orDisjunction.add(Restrictions.gt("m.endDate", new Date()));
+		return (List<Cohort>) sessionFactory.getCurrentSession().createCriteria(Cohort.class)
+				.add(Restrictions.eq("voided", voided))
+				.createAlias("members", "m")
+				.add(Restrictions.le("m.startDate", new Date()))
+				.add(orDisjunction)
+				.createAlias("m.patient", "p")
+				.add(Restrictions.eq("p.patientId", patientId))
```

Comment:
Hmm I think it maybe because I'm not saving it in the db. I will look into this.

Also, I think I know why the comments are jumbled. Your newer comments aren't being shown in the conversation view. Adding new comments in the Files tab seems to be messing up with the order. For me, the conversation view is easier to follow since it shows comments in chronological order. Let me know, if you feel the same.